### DISCUSSION TITLE
Github Actions for Preview/Beta packages?

### DISCUSSION THREAD

**Discussion by MikeBairdRocks** [PiranhaCMS/piranha.core#1437] [Ideas] [answered]
I was looking into using the latest 9.0 beta (for net5.0) and I don't think it is released to nuget yet. Why not change the Github actions to always publish a preview package?

**Comment by tidyui** [comment, marked as answer]
Do you mean to automatically publish preview packages on push to master or more of a nightly build thing?

Both of these could of course be done if we were to change our workflow a bit to never push/merge anything into master that isn’t halfway done 😆 On the other hand I really don’t like the way this messes up the public NuGet feed. 

Maybe we could utilize the repository package feed here on GitHub for pre beta packages and move to NuGet when we want to release beta/rc packages?

**Reply by MikeBairdRocks** [reply]
Yeah, I think the github packages would work. I just think nightly builds are useful for OSS projects so developers can report bugs quicker and you get a faster feedback loop going with new features.  Also reduces the barrier of having to clone the project and build it yourself.

I mean you could change the workflow too and make the master the production build 😁. It wouldn't be too much effort. The one thing that sticks out would be trying to automate the versioning. Using something like GitVersion or something similar to look at branches and tags.

**Reply by tidyui** [reply]
Yeah, starting from 9.0 we will change our branching strategy so master is production branch, and instead use a dev branch for the upcoming version. It has proven cumbersome after we’ve automated our build process as it can’t be used as soon as we start to work on the next version
